### PR TITLE
Add signals to non nested multigrid

### PR DESCRIPTION
**Pull request by fdrmrc**
This PR adds signals that can be triggered before and after calls to `evaluate_and_process()` or `process_and_evaluate()` which happen internally during prolongation/restriction.

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 1]
```diff
@@ -52,6 +52,53 @@ namespace RepartitioningPolicyTools
...
+    boost::signals2::signal<void(const bool before)> evaluation_restriction;
+
+    /**
+     * This signal is triggered before and after the call to
+     * RemotePointEvaluation::evaluate_and_process() used in
+     * MGTwoLevelTransferNonNested::prolongate_and_add(). The difference
+     * with the @p evaluation_prolongation signal is that also the
+     * communication phase is included.
+     */
+    boost::signals2::signal<void(const bool before)> evaluate_and_process;
```
Comment (peterrum) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:90]:
```suggestion
    boost::signals2::signal<void(const bool before)> prolongation;
```

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 2]
```diff
@@ -52,6 +52,53 @@ namespace RepartitioningPolicyTools
...
+   *
+   */
+  struct SignalsNonNested
+  {
+    /**
+     * This signal is triggered before and after the call to the actual
+     * evaluationfunction inside RemotePointEvaluation::evaluate_and_process()
+     * during prolongation.
+     */
+    boost::signals2::signal<void(const bool before)> evaluation_prolongation;
```
Comment (peterrum) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:74]:
```suggestion
    boost::signals2::signal<void(const bool before)> prolongation_cell_loop;
```

**Review by peterrum — APPROVED** [review]

File: tests/multigrid-global-coarsening/events_non_nested.output [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (peterrum) [inline tests/multigrid-global-coarsening/events_non_nested.output:1]: @fdrmrc  It looks like this test depends on trilinos. Could you rename the file!? I'll merge once the tests have passed!

Comment (fdrmrc) [reply tests/multigrid-global-coarsening/events_non_nested.output]: So done, sorry for that

**Pull request merged**